.github/workflows/ubuntu-windows.yml

```
@@ -0,0 +1,401 @@
+        uses: hendrikmuhs/ccache-action@v1
+        with:
+          key: ${{ matrix.os }}-${{ matrix.shared }}-${{ matrix.sirius }}-${{ matrix.xprs }}-${{ matrix.python-version }}
+
+      - name: setup envoronment
+        uses: ./.github/workflows/setup-env
+        with:
+          os: ${{ matrix.os }}
+          python-version: ${{ matrix.python-version }}
+          dotnet: ${{ matrix.dotnet }}
+          python: 'ON'
+
+      - name: Download sirius
+        uses: actions/download-artifact@v2
+        with:
+          name: ${{ matrix.os }}_sirius
+          path: ${{ env.SIRIUS_INSTALL_PATH }}
+
+      - name: Checkout xpressmp813 linux
+        if: ${{ startsWith(matrix.os, 'ubuntu') }}
+        uses: actions/checkout@v2
+        with:
+          repository: rte-france/xpress-mp
+          path: ${{ env.XPRESSDIR }}
+          ref: master
+          token: ${{ secrets.ACCESS_TOKEN }}
+      - name: Checkout xpressmp813 windows
+        if: ${{ startsWith(matrix.os, 'windows') }}
+        uses: actions/checkout@v2
+        with:
+          repository: rte-france/xpress-mp-temp
+          path: ${{ env.XPRESSDIR }}
+          ref: master
+          token: ${{ secrets.ACCESS_TOKEN }}
+
+      - name: set cache variables
+        id: cache
+        shell: bash
+        run: |
+          OS=${{ matrix.os }}
+          [ $OS == "ubuntu-20.04" ] && CCACHE="-DCMAKE_C_COMPILER_LAUNCHER=ccache" || CCACHE=""
+          [ $OS == "ubuntu-20.04" ] && CXXCACHE="-DCMAKE_CXX_COMPILER_LAUNCHER=ccache" || CXXCACHE=""
+          echo "::set-output name=CMAKE_CCACHE::$CCACHE"
+          echo "::set-output name=CMAKE_CXXCACHE::$CXXCACHE"
+
+      - name: Add xpress and sirius bin to path
+        if: ${{ startsWith(matrix.os, 'windows') }}
+        shell: bash
+        run: |
+          echo "${{ env.XPRESS }}" >> $GITHUB_PATH
+          echo "${{ env.SIRIUS_INSTALL_PATH }}\bin" >> $GITHUB_PATH
+
+      - name: Configure ortools
+        shell: bash
+        run: |
+          cmake -S . -B build -DPython3_ROOT_DIR="${{ env.pythonLocation }}" -DBUILD_SAMPLES=OFF -DBUILD_EXAMPLES=OFF -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX="build/install" -DBUILD_SHARED_LIBS=${{ matrix.shared }} -DBUILD_DEPS=ON -DUSE_SIRIUS=${{ matrix.sirius }} -Dsirius_solver_DIR="${{ env.SIRIUS_INSTALL_PATH }}/cmake" -DUSE_XPRESS=${{ matrix.xprs }} -DXPRESS_ROOT="${{ env.XPRESSDIR }}" -DBUILD_PYTHON=${{ matrix.python }} -DBUILD_JAVA=${{ matrix.java }} -DBUILD_DOTNET=${{ matrix.dotnet }} ${{ steps.cache.outputs.CMAKE_CCACHE }} ${{ steps.cache.outputs.CMAKE_CXXCACHE }}
+
+      - name: Build ortools Linux
+        if: ${{ startsWith(matrix.os, 'ubuntu') }}
+        shell: bash
+        run: cmake --build build --config Release --target all install -j2
+
+      - name: Build ortools windows
+        if: ${{ startsWith(matrix.os, 'windows') }}
+        run: cmake --build build --config Release --target ALL_BUILD INSTALL -v -- /maxcpucount
+
+      - name: set ld_library_path
```

```suggestion
      - name: Set LD_LIBRARY_PATH (required to run examples)
```

Question : is LD_LIBRARY_PATH used in Windows ?

---

nope, you're right